foyer/general_forcefield.py

```diff
@@ -0,0 +1,502 @@
+import collections
+import glob
+import os
+from tempfile import NamedTemporaryFile
+import xml.etree.ElementTree as ET
+from pkg_resources import resource_filename
+import warnings
+import re
+from copy import deepcopy
+
+import gmso
+from gmso.external import from_mbuild
+from gmso.core import element
+
+import mbuild as mb
+from foyer.atomtyper import find_atomtypes
+from foyer.exceptions import FoyerError
+from foyer import smarts
+from foyer.utils.io import import_
+from foyer.utils.external import get_ref
+
+
+# Copy from original forcefield.py
+def preprocess_forcefield_files(forcefield_files=None, backend='gmso'):
+    """Pre-process foyer Forcefield XML files"""
+    if forcefield_files is None:
+        return None
+
+    tmp_processed_files = list()
+    if backend == 'gmso':
+        # Run through the forcefield XML conversion
+        from gmso.external.convert_foyer_xml import from_foyer_xml
+        for idx, file in enumerate(forcefield_files):
+            _, suffix = os.path.split(file)
+            tempfile = NamedTemporaryFile(suffix=suffix, delete=False)
+            try:
+                from_foyer_xml(foyer_xml=str(file),
+                          gmso_xml=str(tempfile.name), overwrite=True)
+                tmp_processed_files.append(tempfile.name)
+            except:
+                warnings.warn(f'Could not convert {str(file)}, attempt to read in as is.')
+                tmp_processed_files.append(str(file))
+    else:
+        raise FoyerError('Backend not supported')
+
+    return tmp_processed_files
+
+
+class Forcefield(object):
+    """General Forcefield object that can be created by either a GMSO XML
+    forcefield file of Foyer XML forcefield file
+
+    Parameters
+    ----------
+    forcefield_files : list of str, optional, default=None
+        List of forcefield files to load.
+    name : str, optional, None
+        Name of a forcefield to load that is packaged within foyer.
+    backend : str, optional, default='gmso'
+        Name of the backend used to store all the Types' information.
```

Comment:
Seems like theres a small wording issue with the Types'. Not sure if it should be:
'Types'
or 
Type